crates/l1tx/src/filter/types.rs

```diff
@@ -31,7 +96,10 @@ pub struct TxFilterConfig {
     pub expected_blobs: SortedVec<Buf32>,
 
     /// For deposits that might be spent from.
-    pub expected_outpoints: SortedVec<Outpoint>,
+    pub expected_outpoints: SortedVec<DepositSpendConfig>,
+
+    /// For withdrawal fulfillment transactions sent by bridge operator.
+    pub expected_withdrawal_fulfillments: SortedVec<ExpectedWithdrawalFulfillment>,
```

Comment:
Ah I think I see why those `Ord` impls were there.  These probably should not be `SortedVec`s then.  Or maybe we should define a `SortedMap`/`SortedLookupTable` type that's parametric over some trait impl like:

```rs
trait TableSchema {
    type Item;
    type Key: Eq + PartialEq + Ord + PartialOrd;
    fn get_key(i: &Self::Item) -> &Self::Key;
}
```

And then lets us do searches/lookups based on that `get_key` fn.

We could also theoretically redefine the `DepositsTable` based on this *in theory*, but I wouldn't do that here.